### PR TITLE
Player item inventory

### PR DESCRIPTION
**Pull request by AminArria**
- Adds the concept of an inventory to a player
- Adds a config for `max_inventory_size` of a Character
- Adds `pickup_mechanic` for Loot to define how the pickup should work
- Adds function to use an item in the inventory

### CODE REVIEW

File: native/lambda_game_engine/src/player.rs [hunk 1]
```diff
@@ -272,6 +276,24 @@ impl Player {
...
+                self.inventory[slot_at] = Some(loot.clone());
+                true
+            }
+            None => false,
+        }
+    }
+
+    pub fn inventory_get_at(&mut self, inventory_at: usize) -> Option<Loot> {
+        self.inventory[inventory_at].take()
+    }
```
Comment (mFragaBA) [inline native/lambda_game_engine/src/player.rs:296]: I would expect a `X_get_at` function to get the value and do nothing with self. Maybe `inventory_take_at` is less confusing, what do you think?

File: native/lambda_game_engine/src/game.rs [hunk 2]
```diff
@@ -338,10 +347,15 @@ fn get_next_id(next_id: &mut u64) -> u64 {
...
-            loot.effects
-                .iter()
-                .for_each(|effect| player.apply_effect(effect, EntityOwner::Loot));
-            false
+            match loot.pickup_mechanic {
+                PickupMechanic::CollisionToInventory => player.put_in_inventory(loot),
+                PickupMechanic::CollisionUse => {
+                    loot.effects
+                        .iter()
+                        .for_each(|effect| player.apply_effect(effect, EntityOwner::Loot));
```
Comment (mFragaBA) [inline native/lambda_game_engine/src/game.rs:355]: there's an existing function named `apply_effects` in `player.rs` that does this same thing. Can we re-use it? (i can't add the proper indentation here)
```suggestion
                    player.apply_effects(&loot.effects, EntityOwner::Loot);
```

**Review by mFragaBA — CHANGES_REQUESTED** [review]
Nice work on the inventory. If we eventually add equipable items I think we can extend the current implementation without greater issues. I left two small comments. If it's not too much work can we add some tests for the inventory functionallity?

File: native/lambda_game_engine/src/player.rs (hunk 1 above)
Comment (AminArria) [reply native/lambda_game_engine/src/player.rs]: That makes a lot of sense, renamed in e2ca46f

File: native/lambda_game_engine/src/game.rs (hunk 2 above)
Comment (AminArria) [reply native/lambda_game_engine/src/game.rs]: Missed that when moving this, fixed in [83220fb](https://github.com/lambdaclass/game_backend/pull/38/commits/83220fbf56cb7ca2365e3180f75a915afe0c858c)

**Review by mFragaBA — APPROVED** [review]
LGTM!

**Pull request merged**